lib/mail/version_specific/ruby_1_9.rb

```
@@ -125,7 +125,11 @@ def Ruby19.q_value_decode(str)
     def Ruby19.param_decode(str, encoding)
       str = uri_parser.unescape(str)
       str = charset_encoder.encode(str, encoding) if encoding
-      str
+      decoded = str.encode(Encoding::UTF_8, :invalid => :replace, :replace => "")
+      decoded.valid_encoding? ? decoded : decoded.encode(Encoding::UTF_16LE, :invalid => :replace, :replace => "").encode(Encoding::UTF_8)
+    rescue Encoding::UndefinedConversionError, ArgumentError, Encoding::ConverterNotFoundError
```

Do we have test coverage for these cases?

Should we swallow these errors and return a utf-8 string with invalid encoding?